formwidgets/Blocks.php

```diff
@@ -34,18 +34,67 @@ public function init()
     }
 
     /**
+     * {@inheritDoc}
+     */
+    protected function loadAssets()
+    {
+        $this->addCss('css/blocks.css', 'Winter.Blocks');
+        $this->addJs('js/blocks.js', 'Winter.Blocks');
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function render()
+    {
+        $this->prepareVars();
+        return $this->makePartial('block');
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function onAddItem()
+    {
+        $groupCode = post('_repeater_group');
+
+        $index = $this->getNextIndex();
+
+        $this->prepareVars();
+        $this->vars['widget'] = $this->makeItemFormWidget($index, $groupCode);
+        $this->vars['indexValue'] = $index;
+
+        $itemContainer = '@#' . $this->getId('items');
+        $addItemContainer = '#' . $this->getId('add-item');
+
+        return [
+            $addItemContainer => '',
+            $itemContainer => $this->makePartial('block_item') . $this->makePartial('block_add_item')
+        ];
+    }
+
+    /**
+     * {@inheritDoc}
+     *
      * This method overrides the base repeater processGroupMode to implement block functionality without pre-defining a
      * group.
      */
     protected function processGroupMode(): void
     {
         $definitions = [];
         foreach (BlockManager::instance()->getConfigs($this->config->blockContext ?? null) as $code => $config) {
-            if (in_array($code, $this->ignore)) {
+            if (
+                in_array($code, $this->ignore)
+                || count(array_intersect($config['context'], $this->ignore))
+            ) {
                 continue;
             }
 
-            if (!empty($this->allow) && !in_array($code, $this->allow)) {
+            if (
+                !empty($this->allow)
+                && !in_array($code, $this->allow)
+                && !count(array_intersect($config['context'], $this->allow))
+            ) {
```

Comment:
Is this documented?

Reply:
Not yet - was waiting to see if it would be accepted.

Reply:
Could you document it so I can better understand the API for it?